tuf/repository_lib.py

```diff
@@ -1284,41 +1304,22 @@ def generate_targets_metadata(targets_directory, target_files, version,
 
   # Ensure the user is aware of a non-existent 'target_directory', and convert
   # it to its abosolute path, if it exists.
-  targets_directory = _check_directory(targets_directory)
+  check_exists = not use_existing_fileinfo
+  targets_directory = _check_directory(targets_directory, check_exists)
 
-  # Generate the fileinfo of all the target files listed in 'target_files'.
-  for target, fileinfo in six.iteritems(target_files):
+  if use_existing_fileinfo:
+    for target, fileinfo in six.iteritems(target_files):
 
-    # The root-most folder of the targets directory should not be included in
-    # target paths listed in targets metadata.
-    # (e.g., 'targets/more_targets/somefile.txt' -> 'more_targets/somefile.txt')
-    relative_targetpath = target
+      # Ensure all fileinfo entries in target_files have a non-empty hashes dict
+      if not fileinfo.get('hashes', None):
+        raise securesystemslib.exceptions.Error('use_existing_hashes option set'
+            ' but no hashes exist in roledb for ' + repr(target))
 
-    # Note: join() discards 'targets_directory' if 'target' contains a leading
-    # path separator (i.e., is treated as an absolute path).
-    target_path = os.path.join(targets_directory, target.lstrip(os.sep))
+      filedict[target.replace('\\', '/').lstrip('/')] = fileinfo
```

Comment:
Unrelated: I think this should have happened before (see https://github.com/theupdateframework/tuf/pull/1007#discussion_r399222066)

Reply:
Hopefully it's OK to remove this as we consolidate path handling under #1008, or as a follow on to those changes?

Reply:
Under #1008 or a follow on is okay for me!